### PR TITLE
Scanning Page #5: Add delete unsaved candidates cron job

### PR DESCRIPTION
**Pull request by acrellin**

### CODE REVIEW

File: jobs/delete_unsaved_candidates.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import datetime
+from skyportal.models import init_db, Candidate, DBSession
+from baselayer.app.env import load_env
+
+
+env, cfg = load_env()
+init_db(**cfg["database"])
+
+cutoff_datetime = datetime.datetime.now() - datetime.timedelta(
+    days=cfg["misc"]["days_to_keep_unsaved_candidates"]
```
Comment (stefanv) [inline jobs/delete_unsaved_candidates.py]: No problem here; just a reminder that `cfg["misc.days_to_keep_unsaved_candidates"]` is shorthand for the same.

File: jobs/delete_unsaved_candidates.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+)
+
+c = Candidate.query.filter(Candidate.source_id.is_(None)).filter(
+    Candidate.created_at <= cutoff_datetime
+).delete()
+
+DBSession.commit()
+
+print("Deleted", c, "unsaved candidates.")
```
Comment (stefanv) [inline jobs/delete_unsaved_candidates.py]: f-string?

File: jobs/delete_unsaved_candidates.py (hunk 1 above)
Comment (acrellin) [reply jobs/delete_unsaved_candidates.py]: Done

File: jobs/delete_unsaved_candidates.py (hunk 2 above)
Comment (acrellin) [reply jobs/delete_unsaved_candidates.py]: Done

**Comment by acrellin** [comment]
[ch682]

File: jobs/delete_unsaved_candidates.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    Source.query
+    .filter(Source.is_candidate.is_(True))
+    .filter(Source.is_source.is_(False))
+    .filter(Source.created_at <= cutoff_datetime)
+    .delete()
+)
+
+DBSession.commit()
+
+print(f"Deleted {c} unsaved candidates.")
```
Comment (profjsb) [inline jobs/delete_unsaved_candidates.py:25]: is `c` a number that gets returned? If so, consider naming it `num_deleted` or the like

File: jobs/delete_unsaved_candidates.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import datetime
+from skyportal.models import init_db, Source, DBSession
+from baselayer.app.env import load_env
+
+
+env, cfg = load_env()
+init_db(**cfg["database"])
+
+cutoff_datetime = datetime.datetime.now() - datetime.timedelta(
+    days=cfg["misc.days_to_keep_unsaved_candidates"]
```
Comment (profjsb) [inline jobs/delete_unsaved_candidates.py:12]: perhaps we should check to make sure this config variable is integer and between 1 and X days.

File: jobs/delete_unsaved_candidates.py (hunk 3 above)
Comment (acrellin) [reply jobs/delete_unsaved_candidates.py]: Done

File: jobs/delete_unsaved_candidates.py (hunk 4 above)
Comment (acrellin) [reply jobs/delete_unsaved_candidates.py:12]: Sure, what's a reasonable max? 30 days?

Comment (acrellin) [reply jobs/delete_unsaved_candidates.py]: Done

**Comment by acrellin** [comment]
Closing in favor of an upcoming replacement PR

**Pull request closed without merging**